plugins/redfish/fu-plugin-redfish.c

```diff
@@ -45,23 +46,137 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_redfish_client_discover_uefi_credentials (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	guint32 indications_le;
+	g_autofree gchar *userpass_safe = NULL;
+	g_auto(GStrv) split = NULL;
+	g_autoptr(GBytes) indications = NULL;
+	g_autoptr(GBytes) userpass = NULL;
+
+	/* get the uint32 specifying if there are EFI variables set */
+	indications = fu_efivar_get_data_bytes (REDFISH_EFI_INFORMATION_GUID,
+						REDFISH_EFI_INFORMATION_INDICATIONS,
+						NULL, error);
+	if (indications == NULL)
+		return FALSE;
+	if (g_bytes_get_size (indications) != 4) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INVALID_FILE,
+			     "invalid value for %s, got %" G_GSIZE_FORMAT " bytes",
+			     REDFISH_EFI_INFORMATION_INDICATIONS,
+			     g_bytes_get_size (indications));
+		return FALSE;
+	}
+	memcpy (&indications_le, g_bytes_get_data (indications, NULL), 4);
```

Comment:
isn't this going to matter what architecture it's run on?

Reply:
It is, good catch. Fixed.